File: Data Structures & Algorithms/maximum-subarray/submission-5.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        if not nums:
            return 0
        maxSum = nums[0]
        currSum = 0

        for num in nums:
            currSum = max(num+currSum, num)
            maxSum = max(currSum, maxSum)
        return maxSum

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-5.java
class Solution {
    public int longestConsecutive(int[] nums) {
        int toRet = 0;
        HashSet<Integer> allNums = new HashSet();
        for(int num: nums){
            allNums.add(num);
        }

        for(int num: nums){
            if(!allNums.contains(num-1)){
                int curr = 1;
                while(allNums.contains(num+curr)){
                    curr+=1;
                }
                toRet = Math.max(toRet, curr);
            }
        }
        return toRet;
        
        
    }
}
